src/store/keycodes.js

```diff
@@ -0,0 +1,218 @@
+  store.commit('app/setIso', !isANSI());
+  const keycodes = [
+    ...(isANSI()
+      ? keycodePickerTabLayout.ANSI_ISO
+      : keycodePickerTabLayout.ISO_ANSI),
+    ...keycodePickerTabLayout.special,
+    ...(isSteno ? steno : [])
+  ];
+  const { keycodeLUT } = keymapExtras[getOSKeyboardLayout()];
+  return keycodes.map((keycodeObject) =>
+    toLocaleKeycode(keycodeLUT, keycodeObject)
+  );
+}
+
+/**
+ * Counts the number of potential matches in each keycode array.
+ * The keycode arrays represent the source data used to render the keycodes
+ * display. This count is used to give hints of matches per tab in the UI.
+ *
+ * @param {string} filter
+ * @param {Array.<KeycodeDefinition|KeycodeLabel|WidthPlaceholder>} collection of keycode objects
+ * @returns
+ */
+function countMatches(filter, collection) {
+  filter = filter.toUpperCase();
+  return collection.reduce((matchCount, { code, name, title }) => {
+    if (!isUndefined(code)) {
+      if (
+        code.includes(filter) ||
+        name?.toUpperCase().includes(filter) ||
+        title?.toUpperCase().includes(filter)
+      ) {
+        matchCount += 1;
+      }
+    }
+    return matchCount;
+  }, 0);
+}
+
+/**
+ * Use Options style for now.
+ */
+export const useKeycodesStore = defineStore('keycodes', {
+  /**
+   *
+   * @returns {KeycodeStoreState}
+   */
+  state: () => ({
+    keycodes: [
+      ...keycodePickerTabLayout.ANSI_ISO,
+      ...keycodePickerTabLayout.special
+    ],
+    searchFilter: '',
+    searchCounters: {
+      ANSI: 0,
+      'ISO/JIS': 0,
+      Quantum: 0,
+      KeyboardSettings: 0,
+      AppMediaMouse: 0
+    },
+    steno: false,
+    active: 'ANSI'
+  }),
+  getters: {
+    lookupKeyPressCode: () => (searchTerm) =>
+      this.lookupKeycode(searchTerm, true),
+    lookupKeycode:
+      (state) =>
+      (searchTerm, isKeys = false) =>
+        state.keycodes.find(
+          ({ code, keys }) =>
+            code === searchTerm || (isKeys && keys && keys === searchTerm)
+        )
+  },
+  actions: {
+    changeActive(newActive) {
+      this.active = newActive;
+    },
+    enableSteno() {
+      this.steno = true;
+      this.keycodes = generateKeycodes(getOSKeyboardLayout(), this.steno);
+    },
+    disableSteno() {
+      this.steno = false;
+      this.keycodes = generateKeycodes(getOSKeyboardLayout(), this.steno);
+    },
+    updateKeycodeNames() {
+      this.keycodes = generateKeycodes(getOSKeyboardLayout(), this.steno);
+    },
+    setSearchFilter(newVal) {
+      this.searchFilter = newVal;
+      if (this.searchFilter !== '') {
+        this.searchCounters = {
+          ANSI: countMatches(this.searchFilter, ansi),
+          'ISO/JIS': countMatches(this.searchFilter, iso_jis),
+          Quantum: countMatches(this.searchFilter, quantum),
+          KeyboardSettings: countMatches(this.searchFilter, settings),
+          AppMediaMouse: countMatches(this.searchFilter, media)
+        };
+      }
+    }
```

Comment:
This is a pre-existing issue but the collections passed to `countMatches` (`ansi`, `iso_jis`, etc.) are read-only variables imported from `./modules/keycodes/###`. Consequently, these keycode collections are not localized at all causing incorrect match counts to be reported.

For example, if I search for "ß" with the German OS layout, the `ß` key will be highlighted but the count in both ANSI and ISO tabs will be "(0)". Same story if I search for "Y" with the German OS layout; DE_Y and KC_Y (=DE_Z) are both highlighted but the ANSI tab will only report a single match because it successfully finds a "Y" in the uppercase version of `KC_Y`'s `name` but `KC_Z` aka `DE_Y` has "Z" as its `name` in the `ansi` variable so the `countMatches` does not count it.

The solution would be to map the `toLocalKeycode` on each collection before passing it to the `countMatches`. I'll open a separate issue and open a PR for a possible implementation.

Reply:
Good catch.